homeassistant/components/mysensors.py

```diff
@@ -147,11 +147,11 @@ def gw_start(event):
         tcp_port = gway.get(CONF_TCP_PORT)
         in_prefix = gway.get(CONF_TOPIC_IN_PREFIX)
         out_prefix = gway.get(CONF_TOPIC_OUT_PREFIX)
-        GATEWAYS[device] = setup_gateway(
+        GATEWAYS.append(setup_gateway(
             device, persistence_file, baud_rate, tcp_port, in_prefix,
-            out_prefix)
-        if GATEWAYS[device] is None:
-            GATEWAYS.pop(device)
+            out_prefix))
+        if GATEWAYS[index] is None:
```

Comment:
This is a very weird way to check if `setup_gateway` returned `None` and also one that won't work if it happens twice (because `index` is further ahead of the length of `GATEWAYS`, causing an `IndexError`).

I would suggest the following code:

```python
# Please pick a better name than g_setup
g_setup = setup_gateway(…)
if g_setup is not None:
    GATEWAYS.append(g_setup)
```